src/store/undoRedo/getChangedKeys.ts

```diff
@@ -7,12 +7,23 @@ export function getChangedKeys(a: object, b: object): string[] {
       const aVal = (a as Record<string, unknown>)[key]
       const bVal = (b as Record<string, unknown>)[key]
 
-      if (typeof aVal === 'object' && typeof bVal === 'object' && aVal !== null && bVal !== null) {
-        changedKeys.push(...getChangedKeys(aVal, bVal).map((k) => `${key}.${k}`))
+      if (isObjectOrUndefined(aVal) && isObjectOrUndefined(bVal)) {
+        changedKeys.push(...getChangedKeys(
+          ensureObject(aVal),
+          ensureObject(bVal),
+        ).map((k) => `${key}.${k}`))
       } else if (aVal !== bVal) {
         changedKeys.push(key)
       }
     })
 
   return changedKeys
 }
+
+function ensureObject(obj: any): object {
+  return obj !== null && obj !== undefined ? obj : {}
+}
+
+function isObjectOrUndefined(obj: any): boolean {
```

Comment:
👍

Reply:
@object-required , thanks for creating such methods. I'd suggest we put them under utils folder, eg: `floip\flow-builder\src\utils\check-utils.ts` ? In fact, these would be very useful for other implementations. What do you think?
- [] move utils under the appropriate folder